crates/ironrdp-graphics/src/pointer.rs

```diff
@@ -349,16 +350,23 @@ enum ColorStrideReader {
 }
 
 impl ColorStrideReader {
-    fn new(bpp: u16, stride: Stride) -> Self {
-        match bpp {
+    fn new(bpp: u16, stride: Stride) -> Result<Self, PointerError> {
+        Ok(match bpp {
             1 => Self::Bitmask(BitmaskStrideReader::new(stride)),
             bpp => Self::Color {
-                bpp,
+                bpp: {
+                    // INVARIANT: bpp == 16 || bpp == 24 || bpp == 32
+                    if !SUPPORTED_COLOR_BPP[1..].contains(&bpp) {
+                        return Err(PointerError::NotSupportedBpp { bpp });
+                    }
```

Comment:
style: https://github.com/Devolutions/IronRDP/blob/8fe288400d617ff24c67ec2cc47df020c9fa7229/STYLE.md#invariants

Only use `INVARIANT:` for declaring invariants (in this case, on the field).

suggestion: You can write something like `Enforce the bpp == 16 || bpp == 24 || bpp == 32 invariant.`